### DISCUSSION TITLE
mdnsresponder block port 53 on Mac devices

### DISCUSSION THREAD

**Discussion by github-grid** [rancher-sandbox/rancher-desktop#5371] [Q&A]
Does any1 know why Rancher is using a DNS port on Mac devices? Whenever Rancher is running, the DNS port is used by the Mac service called mdnsresponder.
This is quite problematic when I want to run other apps that require open 53 port.
The solution is to quit Rancher, run an app that needs access to port 53 and then run Rancher again...
It's not a single issue, because more ppl are affected by this problem in my company.

I tried to disable mdnsresponder by command:
`defaults write /System/Library/LaunchDaemons/com.apple.mDNSResponder.plist NoMulticastAdvertisements -bool true` and after computer reboot verified that service is off (`NoMulticastAdvertisements = 1;`), but for some reason, mdnsresponder is still coming back whenever Rancher is running....

Here is the output from `lsof -iTCP:53 -iUDP:53 -n -P` command.

```
mDNSRespo 285 _mdnsresponder   43u  IPv4 0x820d689a9a786def      0t0  UDP *:53
com.cisco 316           root   10u  IPv4 0x820d689a9497ba7f      0t0  UDP
```

**Comment by jandubois** [comment]
Please check https://github.com/rancher-sandbox/rancher-desktop/issues/2854#issuecomment-1235023370 for an explanation and potential workaround

**Reply by github-grid** [reply]
Thank you @jandubois, this fixed my issue!

**Reply by joehorsnell** [reply]
Note that the workaround @jandubois mentions above currently [does not work on M3 Macs](https://github.com/rancher-sandbox/rancher-desktop/issues/2854#issuecomment-1917593505) due to the requirement to use VZ emulation on M3 chips, to work around [this issue](https://github.com/rancher-sandbox/rancher-desktop/issues/5943).

**Comment by Explosion-Scratch** [comment]
I had the same issue with a different software (Cloudflare WARP) where port 53 was already bound, these steps worked well for me:

1. Disable everything in **Sharing in Mac Settings**
2. Kill mDNSResponder (`sudo killall -HUP mDNSResponder`)
3. List all processes on port 53:
```
sudo netstat -vanp tcp | grep ".53 "
netstat -vanp tcp | grep ".53 " | awk '{print $9}' | xargs ps -p
```
4. Kill the offending processes, in this case Fig was causing the issue

**Other apps to kill**:
- Docker, might cause this issue
- Fig (mentioned before)
- mDNSResponder
- Double check your `/etc/hosts` looks like the following:

```
127.0.0.1	localhost
255.255.255.255	broadcasthost
::1 localhost
fe80::1%lo0	localhost
```

<details>
<summary>Set hosts CLI</summary>

Warning, this will reset your /etc/hosts file permanently
```
echo "127.0.0.1 localhost" > /etc/hosts
echo "255.255.255.255   broadcasthost" >> /etc/hosts
echo "::1 localhost" >> /etc/hosts
echo "fe80::1%lo0       localhost" >> /etc/hosts
```

</details>
